Expanded on RichText.md (#222)
pharo-open-documentation/pharo-wiki: PharoProjects/RichText.md

```

If you inspect `text`, you will see that `Hello` is bold, `World` is red, `!` is green.

## TextEmphasis

> The class TextEmphasis defines a set of simple (no font change) changes you can do. See the class side of TextEmphasis for the supported styles of emphasis.

Reviewer: ```suggestion
The class `TextEmphasis` defines a set of simple (no font change) changes you can do. See the class side of `TextEmphasis` for the supported styles of emphasis.
```

> It i possible to overlap the different emphasis styles:

Reviewer: ```suggestion
It is possible to overlap the different emphasis styles:
```

```Smalltalk

> text := 'Hello World!' asText.text	addAttribute: TextEmphasis italic from: 7 to: 11;	addAttribute: TextEmphasis struckOut from: 7 to: 11.

Reviewer: ```suggestion
text := 'Hello World!' asText.
text
    addAttribute: TextEmphasis italic from: 7 to: 11;
    addAttribute: TextEmphasis struckOut from: 7 to: 11.
```

```

## Colors
Colored text is done as shown above. Examine the class `Color` for other colors.

## Different fonts
Loading and managing fonts is an issue in itself.

> You can see which fonts you have available in your image using 'StrikeFont actualFamilyNames'.

Reviewer: ```suggestion
You can see which fonts you have available in your image using `StrikeFont actualFamilyNames`.
```

Assume you have a font named: `Bitmap Source Sans Pro` (was the case on Macbook Pro, Pharo 8, fall 2019). 

You need to turn the font into a text attribute:

```Smalltalk

> largeAttribute := TextFontReference		toFont:			(StrikeFont				familyName: 'Bitmap Source Sans Pro'				pointSize: 20).'My larger text' asText addAttribute: largeAttribute from: 4 to: 10.

Reviewer: ```suggestion
largeAttribute := TextFontReference
	toFont: (StrikeFont familyName: 'Bitmap Source Sans Pro' pointSize: 20).

'My larger text' asText addAttribute: largeAttribute from: 4 to: 10.
```

```

> This should give you 'My larger text' with 'larger' being in font size 20 and in the 'Bitmap Source Sans Pro' font.

Reviewer: ```suggestion

This should give you 'My larger text' with 'larger' being in font size 20 and in the 'Bitmap Source Sans Pro' font.
```

## Clickable text.
It is possible to make text do stuff when clicked. This is obviously a huge area. One simple example is:

```Smalltalk
textAction := TextAction new 
	actOnClickBlock: [ 
		Transcript nextPutAll: 'action clicked';cr;endEntry

> 		].'My action text' asText addAttribute: textAction from: 4 to: 10.

Reviewer: ```suggestion
		].

'My action text' asText addAttribute: textAction from: 4 to: 10.
```
